test/kaniko_task_test.go

```diff
@@ -175,19 +242,69 @@ func TestKanikoTaskRun(t *testing.T) {
 	}
 	podName := cluster.PodName
 	pods := c.KubeClient.Kube.CoreV1().Pods(namespace)
-	t.Logf("Retrieved pods for podname %s: %s\n", podName, pods)
+	logger.Infof("Retrieved pods for podname %s: %s\n", podName, pods)
 
-	req := pods.GetLogs(podName, &corev1.PodLogOptions{})
+	// get the logs for the kaniko step
+	req := pods.GetLogs(podName, &corev1.PodLogOptions{Container: "build-step-kaniko"})
 	readCloser, err := req.Stream()
 	if err != nil {
 		t.Fatalf("Failed to open stream to read: %v", err)
 	}
-	defer readCloser.Close()
-	var buf bytes.Buffer
-	out := bufio.NewWriter(&buf)
-	_, err = io.Copy(out, readCloser)
-	if !strings.Contains(buf.String(), kanikoBuildOutput) {
-		t.Fatalf("Expected output %s from pod %s but got %s", kanikoBuildOutput, podName, buf.String())
+	bs, err := ioutil.ReadAll(readCloser)
+	readCloser.Close()
+	if err != nil {
+		t.Fatalf("Failed to read build-step-kaniko log stream: %v", err)
+	}
+	initLogs := string(bs)
```

Comment:
Now that i think, i like 1. Or else, we might ignore the warning and its easier to not fix it.
Its easy to make the check optional later.

Reply:
1. For sure, I should have done this in the first place. I took a slightly different approach but I think you'll like it.

2. I agree, I was worried about flakiness when I started working on it. One thought I had that could be investigated is that `GetLogs` allows following the logs of a container. I _think_ with that, as soon as the pod is created we could grab a handle to all of the containers log streams and build up the logs as they run. There could be a small race condition here but I think it would still be less than what could happen with the current behaviour.